subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2865,6 +2887,10 @@ int ztls_getsockopt_ctx(struct tls_context *ctx, int level, int optname,
 		err = tls_opt_dtls_handshake_timeout_get(ctx, optval,
 							 optlen, true);
 		break;
+
+	case TLS_DTLS_CONNECTION_ID:
```

Comment:
I think this should be a separate socket option, `TLS_DTLS_PEER_CONNECTION_ID` for example. Currently the behaviour of the option is inconsistent (set self CID on write, peer CID on read, also the data format is different in both cases).